bin/install-pr

```diff
@@ -9,63 +9,138 @@
 
 set -eo pipefail
 
+# Read script flags and arguments
+while :
+do
+  case $1 in
+    -h|--help)
+      echo "Install Linkerd with the changes made in a GitHub Pull Request."
+      echo ""
+      echo "Note:"
+      echo "    If installing into an existing Kubernetes cluster, this script"
+      echo "    assumes the cluster is not a KinD cluster."
+      echo ""
+      echo "Usage:"
+      echo "    # Install Linkerd into the 'kubectl' current cluster"
+      echo "    bin/install-pr ####"
+      echo ""
+      echo "    # Install Linkerd into a new KinD cluster"
+      echo "    bin/install-pr [-k|--kind] ####"
+      exit 0
+      ;;
+    -k|--kind)
+      is_kind=1
+      ;;
+    -?*)
+      echo "Error: Unknown option: $1" >&2
+      exit 1
+      ;;
+    *)
+      break
+  esac
+  shift
+done
+
 pr=$1
 
 if [ -z "$pr" ]
 then
-  echo "usage: ${0##*/} (PR number)" >&2
+  echo "Usage: ${0##*/} (PR number)" >&2
   exit 1
 fi
 
 if [ -z "$GITHUB_TOKEN" ]
 then
   # shellcheck disable=SC2016
-  echo 'Generate a personal access token at https://github.com/settings/tokens and set it in the $GITHUB_TOKEN env var'
+  echo 'Error: Generate a personal access token at https://github.com/settings/tokens and set it in the $GITHUB_TOKEN env var'
   exit 1
 fi
 
-auth="Authorization: token $GITHUB_TOKEN"
-
-branch=$(curl -s -H "$auth" "https://api.github.com/repos/linkerd/linkerd2/pulls/$pr" | jq -r '.head.ref')
-
-artifacts=$(curl -s -H "$auth" "https://api.github.com/repos/linkerd/linkerd2/actions/workflows/kind_integration.yml/runs?branch=$branch" | jq -r '.workflow_runs[0].artifacts_url')
+linkerd2_pulls_url="https://api.github.com/repos/linkerd/linkerd2/pulls"
+linkerd2_kind_integration_url="https://api.github.com/repos/linkerd/linkerd2/actions/workflows/kind_integration.yml"
 
-archive_url=$(curl -s -H "$auth" "$artifacts" | jq -r '.artifacts[0].archive_download_url')
+# Get the URL for downloading the artifacts archive
+auth="Authorization: token $GITHUB_TOKEN"
+branch=$(curl -sL -H "$auth" "$linkerd2_pulls_url/$pr" | jq -r '.head.ref')
+artifacts=$(curl -sL -H "$auth" "$linkerd2_kind_integration_url/runs?branch=$branch" | jq -r '.workflow_runs[0].artifacts_url')
+archive=$(curl -sL -H "$auth" "$artifacts" | jq -r '.artifacts[0].archive_download_url')
 
+bindir=$( cd "${BASH_SOURCE[0]%/*}" && pwd )
 dir=$(mktemp -d -t "linkerd-pr-$pr.XXXXXXXXXX")
 
+# Change directory temporarily and download artifact
 cd "$dir" || exit
 
-echo "### Downloading images ###"
-
-curl -L -o archive.zip -H "$auth" "$archive_url"
-
+curl -L -o archive.zip -H "$auth" "$archive"
 unzip -o archive.zip
 
-echo "### Pushing images ###"
+# Push the Linkerd CLI
+image=$(docker load -i cli-bin.tar | sed 's/Loaded image: //')
+docker push "$image"
```

Comment:
do we actually need to push the cli-bin image?

Reply:
Nope I'll remove that!